agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/rest/EndpointsListServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.dke.data.agrirouter.impl.messaging.rest;
+
+import agrirouter.request.Request;
+import agrirouter.request.payload.account.Endpoints;
+import com.dke.data.agrirouter.api.dto.encoding.EncodeMessageResponse;
+import com.dke.data.agrirouter.api.dto.onboard.OnboardingResponse;
+import com.dke.data.agrirouter.api.enums.TechnicalMessageType;
+import com.dke.data.agrirouter.api.env.Environment;
+import com.dke.data.agrirouter.api.factories.impl.EndpointsListMessageContentFactory;
+import com.dke.data.agrirouter.api.service.messaging.EndpointsListService;
+import com.dke.data.agrirouter.api.service.messaging.encoding.EncodeMessageService;
+import com.dke.data.agrirouter.api.service.parameters.EndpointsListParameters;
+import com.dke.data.agrirouter.api.service.parameters.MessageHeaderParameters;
+import com.dke.data.agrirouter.api.service.parameters.PayloadParameters;
+import com.dke.data.agrirouter.api.service.parameters.SendMessageParameters;
+import com.dke.data.agrirouter.impl.EnvironmentalService;
+import com.dke.data.agrirouter.impl.messaging.encoding.EncodeMessageServiceImpl;
+import com.dke.data.agrirouter.impl.validation.ResponseValidator;
+import java.util.Collections;
+import java.util.UUID;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class EndpointsListServiceImpl extends EnvironmentalService
+    implements EndpointsListService, MessageSender, ResponseValidator {
+
+  private EncodeMessageService encodeMessageService;
+  public EndpointsListServiceImpl(Environment environment) {
+    super(environment);
+    encodeMessageService = new EncodeMessageServiceImpl();
+  }
+
+  @Override
+  public String send(EndpointsListParameters parameters) {
+
+    // there is no validation required
+    // parameters.validate();
+
+    EncodeMessageResponse encodedMessage = encodeMessage(parameters);
+
+    SendMessageParameters sendMessageParameters = new SendMessageParameters();
+    sendMessageParameters.onboardingResponse = parameters.onboardingResponse;
+    sendMessageParameters.setEncodedMessages(
+        Collections.singletonList(encodedMessage.getEncodedMessage()));
+    
+
+    sendMessage(sendMessageParameters);
+
+    return encodedMessage.getApplicationMessageID();
+  }
+
+  private EncodeMessageResponse encodeMessage(EndpointsListParameters parameters) {
+
+    String applicationMessageID = UUID.randomUUID().toString();
```

Comment:
Please use the `MessageIdService` for generating message IDs.

Reply:
Done